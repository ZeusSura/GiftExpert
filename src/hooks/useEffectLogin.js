import { useState, useEffect } from "react";
import { callFechApi } from "../helpers/HelperGIFAPI";
export const useFetchGift = (categoria) => {
  const [state, setstate] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    callFechApi(categoria).then((imgs) => {
        setstate({
          data: imgs,
          loading: false,
        });
    });
  }, [categoria]);
  return state;
};
