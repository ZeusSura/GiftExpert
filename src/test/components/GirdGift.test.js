import { shallow } from "enzyme";
import React from 'react'
import GridGift from "../../components/GirdGift";
import { useFetchGift } from "../../hooks/useFecthGift";
jest.mock('../../hooks/useFecthGift')

describe('Evaluacion del componente gridGift', () => {
    
    const gifts=[{
        id:1,
        url:'/123/diasdi.jpg',
        title:'Imagen 1'
    },
    {
        id:2,
        url:'/123/diasdi.jpg',
        title:'Imagen 1'
    },{
        id:3,
        url:'/123/diasdi.jpg',
        title:'Imagen 1'
    },
    {
        id:4,
        url:'/123/diasdi.jpg',
        title:'Imagen 1'
    }]
    const categoria = "Dragon Ball"
    let  wrapper ;
    beforeEach(()=>{
        jest.clearAllMocks()

        
        useFetchGift.mockReturnValue({
            data: gifts,
            loading: true,
        })
        wrapper =shallow(<GridGift categoria={categoria} />)
    })
    test('componente', () => {
        
      
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GridGiftItem').exists()).toBe(true)
        expect(wrapper.find('GridGiftItem').length).toBe(gifts.length)

    })

})
