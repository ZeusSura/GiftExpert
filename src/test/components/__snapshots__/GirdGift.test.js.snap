// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluacion del componente gridGift componente 1`] = `
<Fragment>
  <p>
    cargando...
  </p>
  <h3>
    Dragon Ball
  </h3>
  <div
    className="contenedor-imagen"
  >
    <GridGiftItem
      imagen={
        Object {
          "id": 1,
          "title": "Imagen 1",
          "url": "/123/diasdi.jpg",
        }
      }
      key="1"
    />
    <GridGiftItem
      imagen={
        Object {
          "id": 2,
          "title": "Imagen 1",
          "url": "/123/diasdi.jpg",
        }
      }
      key="2"
    />
    <GridGiftItem
      imagen={
        Object {
          "id": 3,
          "title": "Imagen 1",
          "url": "/123/diasdi.jpg",
        }
      }
      key="3"
    />
    <GridGiftItem
      imagen={
        Object {
          "id": 4,
          "title": "Imagen 1",
          "url": "/123/diasdi.jpg",
        }
      }
      key="4"
    />
  </div>
</Fragment>
`;
